feature/install-modules-and-configure-webpack

whaberek opened the pull request:
Working configure webpack and first page

Commit:
feature/install-modules-and-configure-webpack
Working configure webpack and first page

mtrzesniowski commented on package.json:
```diff
@@ -0,0 +1,79 @@
+    "angular-material": "1.1.0-rc.5",
+    "angular-messages": "1.5.8",
+    "angular-simple-logger": "0.1.7",
+    "angular-ui-router": "0.2.18",
```
You can use newer version of `angular-ui-router` which will provide you an option to specify which component should be in which state

mtrzesniowski commented:
For the structure of the project please leave only `AppComponent` (without other modules)

mtrzesniowski commented on app/components/app/app.component.js:
```diff
@@ -0,0 +1,9 @@
+}
+
+const AppComponent = {
+    controller: AppController,
```
If controller is empty then we don't need it

mtrzesniowski commented on app/components/app/app.component.js:
```diff
@@ -0,0 +1,9 @@
+    template: require('./app.html')
+};
+
+export default AppComponent;
```
Add a new line at the end of the file

mtrzesniowski commented on app/components/app/app.html:
```diff
@@ -0,0 +1,5 @@
+    <div ui-view="navbar"></div>
+    <div ui-view="main"></div>
+    <div ui-view="footer"></div>
+</div>
```
Add a new line at the end of the file

mtrzesniowski commented on app/components/app/app.scss:
```diff
@@ -0,0 +1,13 @@
+
+}
+
+
```
Too much whitespace at the end, please delete

mtrzesniowski commented on app/components/app/index.js:
```diff
@@ -0,0 +1,8 @@
+    .component('appComponent', AppComponent)
+    .name;
+
+export default appModule;
```
Add a new line at the end of the file

Commit:
develop
Structure of the project with only AppComponent

Commit:
develop
Working change after code review (delete empty controller / change version ui-route / add new line in end document)

mtrzesniowski commented on app/components/index.js:
```diff
@@ -0,0 +1,9 @@
+    ])
+    .name;
+
+export default componentsModule;
```
Add a new line at the end of the file

mtrzesniowski commented on app/index.js:
```diff
@@ -0,0 +1,53 @@
+import translateModule from './translate';
+import localStorage from 'angular-local-storage';
+
+
```
Too much whitespace, please delete one line

mtrzesniowski commented on app/index.js:
```diff
@@ -0,0 +1,53 @@
+
+const root = angular
+    .module('app', [
+        uiRouter,         // 3rd party modules
```
Maybe instead of commenting it on the right just put a comment above and after whole block of 3rd party vendors create new line and another one with comment of App modules?

mtrzesniowski commented on app/index.js:
```diff
@@ -0,0 +1,53 @@
+                abstract: true,
+                views: {
+                    'navbar@': {
+                        template: '<navbar-component></navbar-component>'
```
You can change that to `component` routing, see docs for UI-router 1.0

mtrzesniowski commented on app/index.js:
```diff
@@ -0,0 +1,53 @@
+                        template: '<navbar-component></navbar-component>'
+                    },
+                    'footer@': {
+                        template: '<footer-component></footer-component>'
```
You can change that to component routing, see docs for UI-router 1.0

mtrzesniowski commented on app/pages/main/index.js:
```diff
@@ -0,0 +1,24 @@
+                url: '/',
+                views: {
+                    'main@': {
+                        template: '<main-page-component></main-page-component>'
```
You can change that to component routing, see docs for UI-router 1.0

mtrzesniowski commented on app/pages/main/index.js:
```diff
@@ -0,0 +1,24 @@
+    .name;
+
+export default mainPageModule;
+    
```
Too much whitespace at the end of the file - just one line needed

mtrzesniowski commented on app/pages/main/main-page.html:
```diff
@@ -0,0 +1,7 @@
+            <h2 ng-bind="'projectName'| translate"></h2>
+        </div>
+    </md-toolbar>
+</div>
```
Add new line at the end of the file

mtrzesniowski commented on app/pages/main/main-page.html:
```diff
@@ -0,0 +1,7 @@
+<div class="MainPage" ng-cloak>
+    <md-toolbar>
+        <div layout="row" class="toolbar-content">
+            <h2 ng-bind="'projectName'| translate"></h2>
```
We don't use any translations

mtrzesniowski commented on app/pages/main/main.service.js:
```diff
@@ -0,0 +1,11 @@
+
+    }
+
+    $onInit() {
```
`$onInit()` is a controller lifecycle hook, not a service one

whaberek replied:
This was nowhere to be used so I will remove "main.service"

mtrzesniowski commented on app/style/sass/_colors.scss:
```diff
@@ -0,0 +1,2 @@
+$pgs-orange: #f26f21;
+$pgs-black: #231F20;
```
Add newline at the end of the file

mtrzesniowski commented on app/style/sass/_fonts.scss:
```diff
@@ -0,0 +1 @@
+$fonts-main: Arial, Helvetica, sans-serif;
```
Add newline at the end of the file

mtrzesniowski commented on app/style/sass/common.scss:
```diff
@@ -0,0 +1,14 @@
+  li {
+    display: inline-block;
+  }
+}
```
Add newline at the end of the file

mtrzesniowski commented on app/translate/index.js:
```diff
@@ -0,0 +1,18 @@
+import ngTranslate from 'angular-translate';
```
We don't use translations in the project

mtrzesniowski commented on config/config.js:
```diff
@@ -0,0 +1,17 @@
+    },
+    jsBundleName: '[name]-[hash].bundle.js',
+    cssBundleName: '[name]-[hash].css'
+};
```
Add new line at the end of the file (and everywhere where its needed)

mtrzesniowski commented on config/config.js:
```diff
@@ -0,0 +1,17 @@
+
+module.exports = {
+    port: 80,
+    apiUrl :'https://project-5613440220430148247.firebaseio.com',
```
Maybe it's better to create constants provider of angular ?

Commit:
develop
Use new method for ui-router / add new line at the end of files / add navbar and footer

mtrzesniowski commented on app/index.js:
```diff
@@ -0,0 +1,47 @@
+        componentsModule,
+        pagesModule
+    ])
+    .constant("categories", "https://project-5613440220430148247.firebaseio.com/api/v1/categories.json")
```
You should specify only base url of the project (that you will use in the future in making API calls), not an endpoint

mtrzesniowski commented on app/index.js:
```diff
@@ -0,0 +1,47 @@
+        });
+
+        localStorageServiceProvider
+            .setPrefix('framapp');
```
No need for putting `.setPrefix` in new line

mtrzesniowski commented on app/pages/main/index.js:
```diff
@@ -0,0 +1,21 @@
+                    'main@': 'mainPageComponent'
+                }
+            });
+
```
Whitespace to delete

Commit:
develop
Simple change port and path to my database

Commit:
develop
Repair simple bugfix

Commit:
develop
Common BE for all projects
